hw/ip/sram_ctrl/doc/_index.md

```diff
@@ -103,35 +121,55 @@ This mechanism is part of the [life cycle]({{< relref "hw/ip/lc_ctrl/doc" >}}) s
+As opposed to the intra-word diffusion S&P networks, this particular address scrambling network additionally XOR's in a nonce that has the same width as the address.
 
 ### Integrity Error Handling
 
-The `prim_ram_1p_scr` primitive contains an input to indicate whether upstream logic has detected an integrity error (The generation of the integrity error is determined by system integration).
-When an integrity error is encountered, the prim memory module does the following:
-*  Latch the error condition until reset.
+When an integrity error is encountered, the `sram_ctrl` will latch the integrity error send out a `fatal_bus_integ_error` until the next reset (the generation of the integrity error is determined by system integration).
+In addition, this the latched error condition is fed into the `prim_ram_1p_scr` primitive via a dedicated input, causing the scrambling primitive to do the following:
 *  Reverse the nonce used during the address and CTR scrambling.
 *  Disallow any transaction (read or write) on the actual memory macro.
 
 This behavior, combined with other top level defenses, form a multi-layered defense when integrity errors are seen in the system.
 
+### LFSR Initialization Feature
+
+Note that since the scrambling device uses a block cipher in CTR mode, it is undesirable to initialize the memory with all-zeros from a security perspective, as that would reveal the XOR keystream.
+To this end, the `sram_ctrl` contains an LFSR-based initialization mechanism that overwrites the the entire memory with pseudorandom data.
+
+Initialization can be triggered via the {{< regref "CTRL.INIT" >}} CSR, and once triggered, the LFSR is first re-seeded with the nonce that has been fetched together with the scrambling key.
+Then, the memory is initialized with pseudorandom data pulled from the LFSR.
+For each pseudorandom 32bit word, the initialization mechanism computes the corresponding integrity bits and writes both the data and integrity bits (39bit total) through the scrambling device using the most recently obtained scrambling key.
+
+If SW triggers the scrambling key renewal and LFSR initialization at the same time (i.e., with the same CSR write operation), the LFSR initialization will be stalled until a new scrambling has been obtained.
+
+### Code Execution from SRAM
+
+The SRAM controller contains an access control mechanism for filtering instruction fetches from the processor.
+Depending on the OTP chicken switch EN_SRAM_IFETCH (see [OTP memory map]({{< relref "hw/ip/otp_ctrl/doc/#direct-access-memory-map" >}})) value, code execution from SRAM is either tied to the life cycle state via the HW_DEBUG_EN function (see [life cycle docs]({{< relref "hw/ip/lc_ctrl/doc/#hw_debug_en" >}})), or it can be enabled / disabled via the {{< regref "EXEC" >}} CSR.
+The different configuration options are listed in the table below:
+
```

Comment:
does it make sense to include this diagram?
https://docs.google.com/drawings/d/1N_zpxu9T4pj_Mf0mz2y2Qa94F6EeS_-Xw8RVbQyMoTc/edit?usp=sharing

does it help visualize things?

Reply:
yes that would be a good idea. let me add that.

Reply:
added the diagram.